[v1.8] NCL-5276 implicit rebuild optimization 2

vibe13 opened the pull request:
@jbartece this is an extension of the work done in https://github.com/project-ncl/pnc/pull/2777, to create a cache to avoid duplicated checks. As we are processing dependencies, it could be very common that in a set of BuildRecords to be verified during an implicit rebuild, the same dependencies are present (e.g. parent upstream poms). This cache aims to avoid making the same queries for the same dependencies. Thanks for the review!

thescouser89 commented:
Change makes sense and it's a :+1:  from me :)

thescouser89 commented:
@vibe13 BTW, make sure you add those optimizations to the changelog wiki. I think the users will love this :)

vibe13 commented:
@thescouser89 thanks for the review and the reminder, added the changes in the wiki (and added some others fixes / improvements which were missing from the list)! :-)

vibe13 commented:
@jbartece @thescouser89  ok thanks for the review and comment! I definitely agree! So wrt the hibernate caching, I am making some experiments already in https://github.com/project-ncl/pnc/pull/2766 :-). WIP as I am trying to configure tests atm. So, if you agree, I could merge this one for the moment and then when we decide what to do with Hibernate caching (and if the above spike could look good), I can revert to use it and remove this in memory optimization. Wdyt? Thanks!

jbartece commented:
Sure. Merging
